### PR TITLE
[PHPUnit] Add DependsAnnotationWithValueToAttributeRector

### PR DESCRIPTION
**Pull request by samsonasik**
Add `DependsAnnotationWithValueToAttributeRector`, current functionality will only cover current class method, for external, it needs another PR iteration.

### CODE REVIEW

File: src/Rector/ClassMethod/DependsAnnotationWithValueToAttributeRector.php [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+ * @see \Rector\PHPUnit\Tests\Rector\ClassMethod\DependsAnnotationWithValueToAttributeRector\DependsAnnotationWithValueToAttributeRectorTest
+ */
+final class DependsAnnotationWithValueToAttributeRector extends AbstractRector
+{
+    public function __construct(
+        private readonly TestsNodeAnalyzer $testsNodeAnalyzer,
+        private readonly PhpAttributeGroupFactory $phpAttributeGroupFactory,
+        private readonly PhpDocTagRemover $phpDocTagRemover
+    ) {
+    }
```
Comment (samsonasik) [inline src/Rector/ClassMethod/DependsAnnotationWithValueToAttributeRector.php:31]: The dependency injection here seems cause rectify error:

```bash                                                                                
 [ERROR] Could not process                                                      
         "/home/runner/work/rector-phpunit/rector-phpunit/src/Rector/ClassMethod
         /DependsAnnotationWithValueToAttributeRector.php" file, due to:        
         "System error: "Call to undefined method                               
         PhpParser\Internal\TokenStream::haveTagInRange()"                      
         Run Rector with "--debug" option and post the report here: 
https://github.com/rectorphp/rector/issues/new
". On line:
         882                                                                    
```

Comment (samsonasik) [reply src/Rector/ClassMethod/DependsAnnotationWithValueToAttributeRector.php:31]: It seems resolved by directly using string in args on `$this->phpAttributeGroupFactory->createFromClassWithItems()` https://github.com/rectorphp/rector-phpunit/pull/133/commits/4d1244f02e5870084ccbb4e1a59a75b6595d8236

which really strange, but works ...

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

**Review by TomasVotruba — APPROVED** [review]

**Comment by TomasVotruba** [comment]
Thank you

**Pull request merged**